### ISSUE TITLE
Split from buffer view opens a blank buffer

### ISSUE THREAD

**Issue by agoodshort** [nvim-neo-tree/neo-tree.nvim#898]
**Issue:**
When in buffer view, pressing the shortcut `s` on a buffer opens a new blank buffer.

**Expectation:**
Pressing `s` (or any mapping linked to `open_split`) should open the selected buffer in a new window.

**Notes:**
- Pressing `<CR>` opens the buffer as expected
- `s` from filesystem works as expected
- `w` (i.e. `open_with_window_picker`) from buffer view works as expected 
- my plugin configuration: [neo-tree.lua](https://github.com/goodshort/dotfiles/blob/5d61341cdd07f7baeb184c897d8a196df97e5b0e/home/dot_config/exact_nvim/exact_lua/exact_goodshort/exact_plugins/neo-tree.lua)

https://user-images.githubusercontent.com/33832653/234414613-3e1dce08-44ff-43b3-98a8-199806a61216.mov

**Comment by cseickel** [comment]
Confirmed, I get the same behavior.

**Comment by pysan3** [comment]
It seems that it's actually not an empty buffer but a buffer with the name of the buffer id.

So something like `:split 9` is happening instead of `:split <filename>`.

**Comment by pysan3** [comment]
Found the reason.

This line only cares about `:edit` / `:e`: https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8d485aab32da9b8841d4af977f992b82b14af469/lua/neo-tree/utils.lua#L531

But it should do the same thing for `:split` <-> `:sbuffer` I guess?

Not really keen on this part so I don't know how to fix it tho...

**Comment by goodshort** [comment]
Thanks @pysan3 for looking into this and pointing in the right direction.
I have done some testing and I think I found the issue and have a proposal to fix it.

**What's happening:**

- `M.open_file` receives a `bufnr`
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8d485aab32da9b8841d4af977f992b82b14af469/lua/neo-tree/utils.lua#L529

- Afterwards this `bufnr` gets assigned to `bufnr_or_path`
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8d485aab32da9b8841d4af977f992b82b14af469/lua/neo-tree/utils.lua#L543

- Finally we get the below line where we `:vsplit` our passed `bufnr`
https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8d485aab32da9b8841d4af977f992b82b14af469/lua/neo-tree/utils.lua#L609

**What I propose:**

Add some logic to set the buffer to `nil` if using split commands.
We also might want to review the other commands (they can be found [here](https://github.com/nvim-neo-tree/neo-tree.nvim/blob/8d485aab32da9b8841d4af977f992b82b14af469/lua/neo-tree/sources/common/commands.lua#L615), looking up `open_with_cmd` calls)

```lua
if open_cmd == "edit" or open_cmd == "e" then
  -- If the file is already open, switch to it.
  bufnr = bufnr or M.find_buffer_by_name(path)
  if bufnr <= 0 then
    bufnr = nil
  else
    open_cmd = "b"
  end
elseif open_cmd == "vsplit" then
  -- If we want to split, no bufnr should be provided.
  bufnr = nil
end
```
If you agree with the above I am happy to help with a PR.

**Comment by pysan3** [comment]
I would rather go with something like this.

At least in my environment, if we already know the `buf_id`, `:b buf_id` is much faster than `:e filename`, so I think we should try to use `:sb` whenever possible. Is this only for me?

And note that there exists `:sbuffer` for `:split` but no `vsbuffer` kinda thing for `:vsplit`.

```lua
  local cmd_for_buf = { edit = "b", e = "b", split = "sb", sp = "sb" }
  if cmd_for_buf[open_cmd] ~= nil then
    -- If the file is already open, switch to it.
    bufnr = bufnr or M.find_buffer_by_name(path)
    if bufnr <= 0 then
      bufnr = nil
    else
      open_cmd = cmd_for_buf[open_cmd]
    end
  end
```

**Comment by pysan3** [comment]
@cseickel @agoodshort Any updates?

**Comment by agoodshort** [comment]
I have a [branch](https://github.com/agoodshort/neo-tree.nvim/tree/898-split-from-buffer-view-opens-blank-buffer) where I implemented the changes mentioned in my [comment](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/898#issuecomment-1535537683) above. It works well for me and didn't really touch it since...

Let me implement your [suggestion](https://github.com/nvim-neo-tree/neo-tree.nvim/issues/898#issuecomment-1584283057) into it, test it and open a PR.

**Comment by pysan3** [comment]
Thanks @agoodshort!

Would be nice if you could test it against #989 as well since it should be caused by the same reason.

**Comment by agoodshort** [comment]
I tested #989, but issue still persists with the above changes.

**Comment by pysan3** [comment]
@agoodshort Thanks for making the PR.

I commented over there but there's a bug in your code.

When you fix that, #989 should be fixed as well. I mean, your code was not working at all in the first place lol.

Please check.

**Comment by agoodshort** [comment]
Ho alright, just saw the review. Honest mistake 🫣
Funnily enough, this worked for me when I tested it. (ohh actually thinking of it, it might be because of how `Lazy` caches plugins, and still used my previous commit.

Closing this issue since PR was merged.
